snow/engine/snowman/transitive.go

```diff
@@ -473,41 +478,95 @@ func (t *Transitive) Start(ctx context.Context, startReqID uint32) error {
 		return err
 	}
 
+	var (
+		preferredID    ids.ID
+		preferredBlock snowman.Block
+	)
+
+	// Default to using the last accepted block if the VM does not define its
+	// own initial preferred block.
+	if t.getInitialPreferenceVM == nil {
+		preferredID = lastAcceptedID
+		preferredBlock = lastAccepted
+	} else {
+		preferredID, err := t.getInitialPreferenceVM.GetInitialPreference(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get initial preference: %w", err)
+		}
+
+		preferredBlock, err = t.VM.GetBlock(ctx, preferredID)
+		if err != nil {
+			return fmt.Errorf("failed to get initial preferred block: %w", err)
+		}
+	}
+
+	// Check if the initial setup is more recent than the last accepted
+	// block. This can occur if we shut down while a set of blocks was still
+	// processing in consensus.
+	preferredTip := lastAcceptedID
+	if preferredBlock.Height() > lastAccepted.Height() {
+		preferredTip = preferredID
+	}
+
+	preferredChain := make([]snowman.Block, 0)
+	for preferredBlock.Status() == choices.Processing {
+		// During bootstrap our accepted tip can grow, so we should only
+		// re-issue blocks into consensus with a greater height than what
+		// we currently have.
+		if preferredBlock.Height() > lastAccepted.Height() {
```

Comment:
When can this case be false? If `preferredBlock.Height() <= lastAccepted.Height()` doesn't that mean that the block was accepted (and so the status should be `choices.Accepted`)?

Reply:
Can't this happen if the last accepted block grows during bootstrapping so it's now partially through the previously persisted preferred tip? In this case wouldn't we only want to re-issue blocks between our new last accepted tip -> the previously persisted tip instead of all blocks in the previously persisted tip since those would still be processing?